libraries/aws/provisioning/src/aws_iot_provisioning_serializer.c

```diff
@@ -355,30 +381,173 @@ AwsIotProvisioningError_t _AwsIotProvisioning_SerializeCreateKeysAndCertificateR
 
 /*------------------------------------------------------------------*/
 
+AwsIotProvisioningError_t _serializeCertFromCsrPayload( const char * pCertificateSigningRequest,
+                                                        size_t csrLength,
+                                                        IotSerializerEncoderObject_t * pEncoder,
+                                                        bool isDrySerialization )
+{
+    AwsIotProvisioningError_t status = AWS_IOT_PROVISIONING_SUCCESS;
+
+    IotSerializerEncoderObject_t mapEncoder = IOT_SERIALIZER_ENCODER_CONTAINER_INITIALIZER_MAP;
+    IotSerializerError_t serializerResult = IOT_SERIALIZER_SUCCESS;
+    IotSerializerScalarData_t csrData = IotSerializer_ScalarTextStringWithLength( pCertificateSigningRequest,
+                                                                                  csrLength );
+
+    /* Determine the status checking expression logic for the serializer returned
+     * status code based on whether the we are performing a dry-serialization or not.
+     */
+    bool (* isSuccessStatus)( IotSerializerError_t );
+
+    if( isDrySerialization == true )
+    {
+        isSuccessStatus = _checkSuccessOrBufferToSmall;
+    }
+    else
+    {
+        isSuccessStatus = _checkSuccess;
+    }
+
+    /* Encode the payload as a map container. */
+    serializerResult = _pAwsIotProvisioningEncoder->openContainer( pEncoder,
+                                                                   &mapEncoder,
+                                                                   1 );
+
+    if( serializerResult == IOT_SERIALIZER_OUT_OF_MEMORY )
+    {
+        IotLogError( "serializer: Unable to serialize request payload: "
+                     "Failed to allocate memory for map container: Operation={%s}",
+                     CREATE_CERT_FROM_CSR_OPERATION_LOG );
+        status = AWS_IOT_PROVISIONING_NO_MEMORY;
+    }
+    else if( isSuccessStatus( serializerResult ) == false )
+    {
+        IotLogError( "serializer: Unable to serialize request payload: "
+                     "Failed to open map container: Operation={%s}",
+                     CREATE_CERT_FROM_CSR_OPERATION_LOG );
+        status = AWS_IOT_PROVISIONING_INTERNAL_FAILURE;
+    }
+    else
+    {
+        /* Nothing to do here. */
+    }
+
+    if( status == true )
```

Comment:
`status` isn't a `bool`

Reply:
Updated all uses of `bool` to use the status code type